src/ToastStack/ToastStack.tsx

```diff
@@ -0,0 +1,111 @@
+import * as classnames from "classnames";
+import * as PropTypes from "prop-types";
+import * as React from "react";
+import {CSSTransitionGroup} from "react-transition-group";
+import {ToastNotification} from "./ToastNotification";
+
+import {actionPropType} from "./ToastNotification";
+import {ToastType} from "./ToastType";
+
+import "./ToastStack.less";
+
+const propTypes = {
+  className: PropTypes.string,
+  defaultNotificationDurationMS: PropTypes.number,
+  notificationClassName: PropTypes.string,
+  notifications: PropTypes.arrayOf(PropTypes.shape({
+    action: actionPropType,
+    content: PropTypes.node.isRequired,
+    durationMS: PropTypes.number,
+    id: PropTypes.number.isRequired,
+    showCloseButton: PropTypes.bool,
+    type: PropTypes.oneOf(Object.values(ToastType)).isRequired,
+  })),
+  setNotifications: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  defaultNotificationDurationMS: 5000,
+};
+
+const cssClass = {
+  CONTAINER: "ToastStack",
+  NOTIFICATION_WRAPPER: "ToastStack--notificationWrapper",
+  ANIMATION: "ToastStack--animation",
+};
+
+/**
+ * Toast notifications can be used to provide feedback in response to user interaction. The
+ * <ToastStack> renders toast notifications and handles their enter/leave animations.
+ */
+export class ToastStack extends React.PureComponent {
+  static propTypes = propTypes;
+  static defaultProps = defaultProps;
+  static cssClass = cssClass;
+
+  _finiteDuration(notification) {
+    const {defaultNotificationDurationMS} = this.props;
+
+    if (notification.durationMS) {
+      return notification.durationMS !== Infinity;
+    }
+    return defaultNotificationDurationMS !== Infinity;
+  }
+
+  _clearNotification(id) {
+    const {notifications, setNotifications} = this.props;
+
+    setNotifications(notifications.filter(n => n.id !== id));
+  }
+
+  componentDidUpdate(prevProps) {
+    const {defaultNotificationDurationMS} = this.props;
+    const prevNotifications = prevProps.notifications || [];
+    const newNotifications = this.props.notifications || [];
+
+    // If a notification was just added to the stack and it has a finite duration, set up a timer
```

Comment:
Could multiple new notifications be added to the stack at once?

Reply:
Ooh yes great point. This is technically possible, though in practice unlikely given how we use toast notifications. I'll still update the code to handle this case for completeness